Installation issue: py-numpy@1.26.3 with Intel oneAPI (icx) 2025.2.1

### Steps to reproduce the issue

There is a regression in Intel oneAPI 2025.2.1 that prevents building the auto-generated dispatch code in py-numpy. I will create a ticket with Intel to look into this, potentially this is related to a regression discovered when building py-scipy with Intel oneAPI 2025.2.x (and/or 2025.1.y, I need to narrow it down) that, according to the Inte compiler team, will be fixed in Intel oneAPI 2026.

Note: The build system chain spack -> pip -> toml -> meson is astonishingly complex to understand or modify.

```console
$ spack spec -l py-numpy@1.26.4
 -   o4ftduw  py-numpy@1.26.4 build_system=python_pip patches:=873745d arch=linux-almalinux9-x86_64_v3 %c,cxx=oneapi@2025.2.1
[+]  i6k7ldz      ^compiler-wrapper@1.0 build_system=generic arch=linux-almalinux9-x86_64_v3
[e]  4ywflif      ^glibc@2.34 build_system=autotools arch=linux-almalinux9-x86_64_v3
[e]  5zsheof      ^intel-oneapi-compilers@2025.2.1~amd+envmods~nvidia build_system=generic arch=linux-almalinux9-x86_64_v3
[e]  g5tylao      ^intel-oneapi-mkl@2025.2~cluster+envmods~gfortran~ilp64+shared build_system=generic mpi_family=none threads=none arch=linux-almalinux9-x86_64_v3
[+]  35at7gy      ^intel-oneapi-runtime@2025.2.1 build_system=generic arch=linux-almalinux9-x86_64_v3
[e]  svye46d          ^gcc-runtime@11.5.0 build_system=generic arch=linux-almalinux9-x86_64_v3
[+]  3yqagxy      ^py-cython@3.0.12 build_system=python_pip arch=linux-almalinux9-x86_64_v3 %c,cxx=oneapi@2025.2.1
[+]  ncmmc47          ^py-setuptools@69.2.0 build_system=generic arch=linux-almalinux9-x86_64_v3
[+]  wvkzmzd      ^py-meson-python@0.16.0 build_system=python_pip arch=linux-almalinux9-x86_64_v3 %c=oneapi@2025.2.1
[+]  twukedn          ^meson@1.7.0 build_system=python_pip patches:=0f0b1bd arch=linux-almalinux9-x86_64_v3
[+]  qokl6rj          ^ninja@1.12.1+re2c build_system=generic patches:=93f4bb3 arch=linux-almalinux9-x86_64_v3 %c,cxx=oneapi@2025.2.1
[+]  pgqdcpn              ^re2c@3.1 build_system=autotools arch=linux-almalinux9-x86_64_v3 %c,cxx=oneapi@2025.2.1
[+]  f57zrwm          ^py-packaging@25.0 build_system=python_pip arch=linux-almalinux9-x86_64_v3
[+]  ziayij4              ^py-flit-core@3.8.0 build_system=python_pip arch=linux-almalinux9-x86_64_v3
[+]  jhq5e3z          ^py-pyproject-metadata@0.9.1 build_system=python_pip arch=linux-almalinux9-x86_64_v3
[+]  7diwlum      ^py-pip@25.1.1 build_system=generic arch=linux-almalinux9-x86_64_v3
[+]  uyv6hs3      ^py-wheel@0.45.1 build_system=generic arch=linux-almalinux9-x86_64_v3
[+]  d5hqw6k      ^python@3.11.11+bz2~crypt+ctypes+dbm~debug+libxml2+lzma~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches:=13fa8bf,b0615b2,ebdca64,f2fd060 arch=linux-almalinux9-x86_64_v3 %c,cxx=oneapi@2025.2.1
[+]  up5zzka          ^bzip2@1.0.8~debug~pic+shared build_system=generic arch=linux-almalinux9-x86_64_v3 %c=oneapi@2025.2.1
[+]  poyerw6              ^diffutils@3.10 build_system=autotools arch=linux-almalinux9-x86_64_v3 %c=oneapi@2025.2.1
[+]  kritbsu          ^expat@2.7.1~libbsd build_system=autotools arch=linux-almalinux9-x86_64_v3 %c,cxx=oneapi@2025.2.1
[+]  3smxd3p          ^gdbm@1.23 build_system=autotools arch=linux-almalinux9-x86_64_v3 %c=oneapi@2025.2.1
[+]  agmbgh6          ^gettext@0.23.1+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-almalinux9-x86_64_v3 %c,cxx=oneapi@2025.2.1
[+]  itnlir4              ^libiconv@1.18 build_system=autotools libs:=shared,static arch=linux-almalinux9-x86_64_v3 %c=oneapi@2025.2.1
[+]  63ab5yi              ^libxml2@2.13.5~http+pic~python+shared build_system=autotools arch=linux-almalinux9-x86_64_v3 %c=oneapi@2025.2.1
[+]  ajxjz3e              ^tar@1.35 build_system=autotools zip=pigz arch=linux-almalinux9-x86_64_v3 %c=oneapi@2025.2.1
[+]  v6a56ej                  ^pigz@2.8 build_system=makefile arch=linux-almalinux9-x86_64_v3 %c=oneapi@2025.2.1
[+]  di2izlc                  ^zstd@1.5.7+pic+programs build_system=makefile compression:=none libs:=shared,static arch=linux-almalinux9-x86_64_v3 %c,cxx=oneapi@2025.2.1
[+]  gmiener          ^gmake@4.4.1~guile build_system=generic arch=linux-almalinux9-x86_64_v3 %c=gcc@11.5.0
[e]  u7stwk3              ^gcc@11.5.0~binutils+bootstrap~graphite~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++' arch=linux-almalinux9-x86_64_v3
[+]  vo7e56a          ^libffi@3.4.8 build_system=autotools arch=linux-almalinux9-x86_64_v3 %c,cxx=oneapi@2025.2.1
[+]  tiwopgk          ^ncurses@6.5~symlinks+termlib abi=none build_system=autotools patches:=7a351bc arch=linux-almalinux9-x86_64_v3 %c,cxx=oneapi@2025.2.1
[+]  z3ie6zq          ^openssl@3.4.1~docs+shared build_system=generic certs=system arch=linux-almalinux9-x86_64_v3 %c,cxx=oneapi@2025.2.1
[+]  cjabaws              ^perl@5.40.0+cpanm+opcode+open+shared+threads build_system=generic arch=linux-almalinux9-x86_64_v3 %c=oneapi@2025.2.1
[+]  6ywv5vj                  ^berkeley-db@18.1.40+cxx~docs+stl build_system=autotools patches:=26090f4,b231fcc arch=linux-almalinux9-x86_64_v3 %c,cxx=oneapi@2025.2.1
[+]  hpxlhjx          ^pkg-config@0.29.2+internal_glib build_system=autotools arch=linux-almalinux9-x86_64_v3 %c=oneapi@2025.2.1
[+]  gjzv23o          ^readline@8.2 build_system=autotools patches:=1ea4349,24f587b,3d9885e,5911a5b,622ba38,6c8adf8,758e2ec,79572ee,a177edc,bbf97f1,c7b45ff,e0013d9,e065038 arch=linux-almalinux9-x86_64_v3 %c=oneapi@2025.2.1
[+]  5zwms6i          ^sqlite@3.46.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-almalinux9-x86_64_v3 %c=oneapi@2025.2.1
[+]  7tjgfmv          ^util-linux-uuid@2.41 build_system=autotools arch=linux-almalinux9-x86_64_v3 %c=oneapi@2025.2.1
[+]  zfa2hac          ^xz@5.6.3~pic build_system=autotools libs:=shared,static arch=linux-almalinux9-x86_64_v3 %c=oneapi@2025.2.1
[+]  55fizxf          ^zlib@1.3.1+optimize+pic+shared build_system=makefile arch=linux-almalinux9-x86_64_v3 %c,cxx=gcc@11.5.0
[+]  bzyng6j      ^python-venv@1.0 build_system=generic arch=linux-almalinux9-x86_64_v3
```


### Error message

```console
     130    ==> [2025-09-15-11:15:30.950291] '/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/envs/scipy-oneapi-2025.2.1/install/none/none/python-venv-1.0-bzyng6j/bin/python3' '-m' 'pip' '-vvv' '--no-input' '--no-cache-dir' '--disable-pip-version-check' 'install' '--no-deps' '
            --ignore-installed' '--no-build-isolation' '--no-warn-script-location' '--no-index' '--prefix=/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/envs/scipy-oneapi-2025.2.1/install/intel-oneapi-compilers/2025.2.1/py-numpy-1.26.4-o4ftduw' '--config-settings=builddir=bui
            ld' '--config-settings=compile-args=-j4' '--config-settings=setup-args=-Dblas=mkl-dynamic-lp64-seq' '--config-settings=setup-args=-Dlapack=mkl-dynamic-lp64-seq' '.'

     ...

     666            |   ^
     667      ../numpy/core/src/umath/loops.h.src:36:42: note: previous definition is here
     668         36 | typedef struct PyArrayMethod_Context_tag PyArrayMethod_Context;
     669            |                                          ^
     670      2 warnings generated.
     671      [204/494] Compiling C object numpy/core/libloops_arithmetic.dispatch.h_AVX2.a.p/meson-generated_loops_arithmetic.dispatch.c.o
  >> 672    FAILED: numpy/core/libloops_arithmetic.dispatch.h_AVX2.a.p/meson-generated_loops_arithmetic.dispatch.c.o
     673      /home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/envs/scipy-oneapi-2025.2.1/install/none/none/compiler-wrapper-1.0-i6k7ldz/libexec/spack/oneapi/icx -Inumpy/core/libloops_arithmetic.dispatch.h_AVX2.a.p -Inumpy/core -I../numpy/core -Inumpy/core/include -I../numpy/co
            re/include -I../numpy/core/src/common -I../numpy/core/src/multiarray -I../numpy/core/src/npymath -I../numpy/core/src/umath -I/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/envs/scipy-oneapi-2025.2.1/install/intel-oneapi-compilers/2025.2.1/python-3.11.11-d5hqw6k/in
            clude/python3.11 -I/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/cache/build_stage/spack-stage-py-numpy-1.26.4-o4ftduw2hmecpvqn7vvpidaqj626dcrc/spack-src/build/meson_cpu -fcolor-diagnostics -DNDEBUG -Wall -Winvalid-pch -std=c99 -O3 -fno-strict-aliasing -msse -mss
            e2 -msse3 -fPIC -DNPY_INTERNAL_BUILD -DHAVE_NPY_CONFIG_H -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE=1 -D_LARGEFILE64_SOURCE=1 -O3 -DNPY_HAVE_SSE2 -DNPY_HAVE_SSE -DNPY_HAVE_SSE3 -DNPY_HAVE_SSSE3 -DNPY_HAVE_SSE41 -DNPY_HAVE_POPCNT -DNPY_HAVE_SSE42 -DNPY_HAVE_AVX -DNPY_HAVE_F16C -DNPY_H
            AVE_AVX2 -msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mavx -mf16c -mavx2 -DNPY_MTARGETS_CURRENT=AVX2 -MD -MQ numpy/core/libloops_arithmetic.dispatch.h_AVX2.a.p/meson-generated_loops_arithmetic.dispatch.c.o -MF numpy/core/libloops_arithmetic.dispatch.h_AVX2.a.p/meson-genera
            ted_loops_arithmetic.dispatch.c.o.d -o numpy/core/libloops_arithmetic.dispatch.h_AVX2.a.p/meson-generated_loops_arithmetic.dispatch.c.o -c numpy/core/libloops_arithmetic.dispatch.h_AVX2.a.p/loops_arithmetic.dispatch.c
     674      In file included from ../numpy/core/src/umath/loops_arithmetic.dispatch.c.src:15:
     675      In file included from ../numpy/core/src/common/lowlevel_strided_loops.h:3:
     676      In file included from ../numpy/core/src/multiarray/common.h:6:
     677      ../numpy/core/include/numpy/ndarraytypes.h:308:31: warning: redefinition of typedef 'NpyAuxData' is a C11 feature [-Wtypedef-redefinition]
     678        308 | typedef struct NpyAuxData_tag NpyAuxData;

     ...

     693      Stack dump:
     694      0.    Program arguments: /opt/intel/oneapi/compiler/2025.2/bin/compiler/clang @/tmp/icx0504794194AExpXS/icxargl1j8wo
     695      1.    <eof> parser at end of file
     696      2.    Optimizer
     697      3.    Running pass "function(pragma-omp-simd-if,inject-tli-mappings,hir-ssa-deconstruction,hir-temp-cleanup,hir-normalize-casts,hir-propagate-casted-iv,hir-loop-concatenation,hir-pm-symbolic-tripcount-completeunroll,hir-array-transpose,hir-conditional-temp-sinking,hir-loop-peeling,
            hir-early-opt-predicate,hir-lmm,hir-store-result-into-temp-array,hir-aos-to-soa,hir-backward-temp-dependencies-resolver,hir-runtime-dd,hir-mv-const-ub,hir-sinking-for-perfect-loopnest,hir-non-zero-sinking-for-perfect-loopnest,hir-pragma-loop-blocking,hir-loop-distribute-loopnest,hir-
            loop-interchange,hir-generate-mkl-call,hir-loop-blocking,hir-undo-sinking-for-perfect-loopnest,hir-dead-store-elimination,hir-minmax-recognition,hir-identity-matrix-idiom-recognition,hir-lower-small-memset-memcpy,hir-pre-vec-complete-unroll,hir-cond-ldst-motion,hir-dead-store-elimina
            tion,hir-memory-reduction-sinking,hir-lmm,hir-min-max-blob-to-select,hir-last-value-computation,hir-loop-reroll,hir-loop-collapse,hir-loop-fusion,hir-dead-store-elimination,hir-loop-independent-scalar-repl,hir-opt-var-predicate,hir-temp-array-transpose,hir-loop-distribute-memrec,hir-
            loop-reversal,hir-loop-rematerialize,hir-multi-exit-loop-reroll,hir-idiom,hir-if-reversal,hir-unroll-and-jam,hir-mv-variable-stride,hir-opt-predicate,hir-special-opt-predicate,hir-lmm,hir-memory-reduction-sinking,hir-loop-independent-scalar-repl,hir-vec-dir-insert,hir-vplan-vec,hir-v
            plan-vec,hir-post-vec-complete-unroll,hir-general-unroll,hir-scalarrepl-array,hir-nontemporal-marking,hir-prefetching,hir-cg,vpo-cfg-restructuring,vpo-rename-operands,simplifycfg<bonus-inst-threshold=1;no-forward-switch-cond;no-switch-range-to-icmp;no-switch-to-lookup;keep-loops;no-h
            oist-common-insts;no-hoist-loads-stores-with-cond-faulting;no-sink-common-insts;speculate-blocks;simplify-cond-branch;no-invalidate-anders-res;no-speculate-unpredictables>,lower-subscript,sroa<modify-cfg>,nary-reassociate,gvn<>,sroa<modify-cfg>,simplifycfg<bonus-inst-threshold=1;no-f
            orward-switch-cond;no-switch-range-to-icmp;no-switch-to-lookup;keep-loops;no-hoist-common-insts;no-hoist-loads-stores-with-cond-faulting;no-sink-common-insts;speculate-blocks;simplify-cond-branch;no-invalidate-anders-res;no-speculate-unpredictables>,instcombine<max-iterations=1;no-ve
            rify-fixpoint>,loop-carried-cse,dse,addsub-reassoc,vpo-restore-operands,loop-simplify,lower-switch,loop(loop-simplifycfg),vpo-cfg-restructuring)" on module "numpy/core/libloops_arithmetic.dispatch.h_AVX2.a.p/loops_arithmetic.dispatch.c"
     698      4.    Running pass "hir-backward-temp-dependencies-resolver" on function "simd_divide_by_scalar_contig_s64"
  >> 699    icx: error #10105: /opt/intel/oneapi/compiler/2025.2/bin/compiler/clang: core dumped
     700      icx: warning #10102: unknown signal(259129360)
  >> 701      icx: error #10106: Fatal error in /opt/intel/oneapi/compiler/2025.2/bin/compiler/clang, terminated by unknown
     702      [205/494] Compiling C object numpy/core/libloops_arithmetic.dispatch.h_AVX512_SKX.a.p/meson-generated_loops_arithmetic.dispatch.c.o
     703      In file included from ../numpy/core/src/umath/loops_arithmetic.dispatch.c.src:15:
     704      In file included from ../numpy/core/src/common/lowlevel_strided_loops.h:3:
     705      In file included from ../numpy/core/src/multiarray/common.h:6:
     706      ../numpy/core/include/numpy/ndarraytypes.h:308:31: warning: redefinition of typedef 'NpyAuxData' is a C11 feature [-Wtypedef-redefinition]
     707        308 | typedef struct NpyAuxData_tag NpyAuxData;

     ...

     748      ╰─> See above for output.
     749
     750      note: This error originates from a subprocess, and is likely not a problem with pip.
     751      full command: /home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/envs/scipy-oneapi-2025.2.1/install/none/none/python-venv-1.0-bzyng6j/bin/python3 /home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/envs/scipy-oneapi-2025.2.1/install/none/none/py-pip-2
            5.1.1-7diwlum/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py prepare_metadata_for_build_wheel /tmp/tmplp24itjh
     752      cwd: /home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/cache/build_stage/spack-stage-py-numpy-1.26.4-o4ftduw2hmecpvqn7vvpidaqj626dcrc/spack-src
     753      Preparing metadata (pyproject.toml): finished with status 'error'
  >> 754    error: metadata-generation-failed
     755
     756    × Encountered error while generating package metadata.
     757    ╰─> See above for output.
     758
     759    note: This is an issue with the package mentioned above, not pip.
     760    hint: See above for details.

See build log for details:
  /home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/cache/build_stage/spack-stage-py-numpy-1.26.4-o4ftduw2hmecpvqn7vvpidaqj626dcrc/spack-build-out.txt

Traceback (most recent call last):
  File "/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/spack/lib/spack/spack/build_environment.py", line 1184, in _setup_pkg_and_run
    return_value = function(pkg, kwargs)
  File "/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/spack/lib/spack/spack/installer.py", line 2750, in build_process
    return installer.run()
  File "/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/spack/lib/spack/spack/installer.py", line 2630, in run
    self._real_install()
  File "/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/spack/lib/spack/spack/installer.py", line 2715, in _real_install
    phase_fn.execute()
  File "/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/spack/lib/spack/spack/builder.py", line 379, in execute
    self.phase_fn(pkg, pkg.spec, pkg.prefix)
  File "/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/spack/lib/spack/spack/builder.py", line 251, in _adapter
    return phase_fn(spec, prefix)
  File "/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/spack/lib/spack/spack/builder.py", line 156, in __forward
    return self.__getattr__(fn_name)(*args, **kwargs)
  File "/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/spack/lib/spack/spack/builder.py", line 68, in __call__
    return self.phase_fn(self.builder.pkg, spec, prefix)
  File "/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/repos/builtin/repos/spack_repo/builtin/build_systems/python.py", line 420, in install
    pip(*args)
  File "/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/spack/lib/spack/spack/util/executable.py", line 305, in __call__
    raise ProcessError("Command exited with status %d:" % proc.returncode, long_msg)
spack.util.executable.ProcessError: Command exited with status 1:
    '/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/envs/scipy-oneapi-2025.2.1/install/none/none/python-venv-1.0-bzyng6j/bin/python3' '-m' 'pip' '-vvv' '--no-input' '--no-cache-dir' '--disable-pip-version-check' 'install' '--no-deps' '--ignore-installed' '--no-build-isolation' '--no-warn-script-location' '--no-index' '--prefix=/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/envs/scipy-oneapi-2025.2.1/install/intel-oneapi-compilers/2025.2.1/py-numpy-1.26.4-o4ftduw' '--config-settings=builddir=build' '--config-settings=compile-args=-j4' '--config-settings=setup-args=-Dblas=mkl-dynamic-lp64-seq' '--config-settings=setup-args=-Dlapack=mkl-dynamic-lp64-seq' '.'
==> [2025-09-15-11:16:05.640847, 67020] Flagging py-numpy-1.26.4-o4ftduw2hmecpvqn7vvpidaqj626dcrc as failed: ProcessError: Command exited with status 1:
    '/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/envs/scipy-oneapi-2025.2.1/install/none/none/python-venv-1.0-bzyng6j/bin/python3' '-m' 'pip' '-vvv' '--no-input' '--no-cache-dir' '--disable-pip-version-check' 'install' '--no-deps' '--ignore-installed' '--no-build-isolation' '--no-warn-script-location' '--no-index' '--prefix=/home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/envs/scipy-oneapi-2025.2.1/install/intel-oneapi-compilers/2025.2.1/py-numpy-1.26.4-o4ftduw' '--config-settings=builddir=build' '--config-settings=compile-args=-j4' '--config-settings=setup-args=-Dblas=mkl-dynamic-lp64-seq' '--config-settings=setup-args=-Dlapack=mkl-dynamic-lp64-seq' '.'
```

### Information on your system

```
* **Spack:** 1.0.1 (https://github.com/spack/spack/commit/79385e545912dd05008d75bfaca0200c8313b399)
* **Builtin repo:** /home/dom/work/spack-stack/update-to-spack-v1/spst-upd-to-spack-v1/repos/builtin/repos/spack_repo/builtin (50857d8)
* **Python:** 3.9.21
* **Platform:** linux-almalinux9-skylake
```

### Additional information

[py-numpy-1.26.4-error.tar.gz](https://github.com/user-attachments/files/22348151/py-numpy-1.26.4-error.tar.gz)

@adamjstewart  @rgommers

Note. I came up with a workaround and I will create a PR for feedback and comments shortly.

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/py_numpy/package.py
+++ b/repos/spack_repo/builtin/packages/py_numpy/package.py
@@ -176,6 +176,16 @@ class PyNumpy(PythonPackage):
             when="@1.26:1.26.3",
         )
 
+    # https://github.com/spack/spack-packages/issues/1477
+    @when("@1.26 ^intel-oneapi-compilers@2025.2")
+    def patch(self):
+        filter_file(
+            ".get(compiler_id, ['-O3'])",
+            ".get(compiler_id, ['-O1'])",
+            "./numpy/core/meson.build",
+            string=True
+        )
+
     # meson.build
     # https://docs.scipy.org/doc/scipy/dev/toolchain.html#compilers
     conflicts("%gcc@:9.2", when="@2.3:", msg="NumPy requires GCC >= 9.3")

--- a/repos/spack_repo/builtin/packages/py_numpy/package.py
+++ b/repos/spack_repo/builtin/packages/py_numpy/package.py
@@ -183,7 +183,7 @@ class PyNumpy(PythonPackage):
             ".get(compiler_id, ['-O3'])",
             ".get(compiler_id, ['-O1'])",
             "./numpy/core/meson.build",
-            string=True
+            string=True,
         )
 
     # meson.build

